### PR TITLE
Display and/or groups in obsList

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
Sorting is based on groups sorting, so might slightly change from the original order of 'just' observations. Group view works for both AND and OR groups (based on minimumRequired field of a group).

TODO (other PR's):

- 'Edit' groups view
- Create new groups
- Delete groups
- Move observations between groups
- Keep tree state when switching between observations (now collapses all)

Lucuma-schemas release is still processing, so build will fail until then.

### CODE REVIEW

File: explore/src/main/scala/explore/observationtree/obsTree.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import react.primereact.Tree.Node
+
+import java.util.UUID
+import scala.scalajs.js.JSConverters._
+
+sealed trait ObsNode
+
+object ObsNode {
+
+  // given Reusability[ObsNode] = Reusability.byEq
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/observationtree/obsTree.scala:29]: commented code

File: explore/src/main/scala/explore/observationtree/obsTree.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  case class Obs(value: ObsSummary) extends ObsNode
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/observationtree/obsTree.scala:31]: Can we use an `enum` for this?

File: explore/src/main/scala/explore/observationtree/ObsListTree.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import lucuma.ui.reusability.given
+import lucuma.react.table.*
+import lucuma.react.table.ColumnDef
+import org.scalajs.dom.HTMLDivElement
+import explore.observationtree.ObsNode.given
+import explore.components.ui.ExploreStyles
+import react.primereact.Tree
+import lucuma.typed.primereact.treeTreeMod.TreeNodeTemplateOptions
+
+final case class ObsListTree[A](
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/observationtree/ObsListTree.scala:28]: `final` is not necessary anymore for `case class`es in Scala 3.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 4]
```diff
@@ -389,6 +395,16 @@ object ObsTabContents extends TwoPanels:
                         callbacks
         )
       }
-      .render((props, ctx, twoPanelState, resize, layouts, defaultLayout) =>
-        renderFn(props, twoPanelState, defaultLayout, layouts, resize, ctx)
+      .useStreamResourceViewBy((props, _, _, _, _, _) => props.programId) {
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:398]: Can we put this in `ProgramSummaries` and have it cached?

**Review by rpiaggio — APPROVED** [review]
LGTM.

We can move the groups to the cache in a future PR.

File: explore/src/main/scala/explore/observationtree/ObsListTree.scala (hunk 3 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/ObsListTree.scala:28]: I blame Metals for adding it :)

File: explore/src/main/scala/explore/observationtree/obsTree.scala (hunk 1 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/obsTree.scala]: Good catch

File: explore/src/main/scala/explore/observationtree/obsTree.scala (hunk 2 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/obsTree.scala]: Yes, refactored it

**Pull request merged**